modules/pubmaticBidAdapter.js

```diff
@@ -1096,324 +597,85 @@ export const spec = {
-      }
-    });
-
-    if (payload.imp.length == 0) {
-      return;
-    }
-
-    payload.site.publisher.id = conf.pubId.trim();
-    publisherId = conf.pubId.trim();
-    payload.ext.wrapper = {};
-    payload.ext.wrapper.profile = parseInt(conf.profId) || UNDEFINED;
-    payload.ext.wrapper.version = parseInt(conf.verId) || UNDEFINED;
-    // TODO: fix auctionId leak: https://github.com/prebid/Prebid.js/issues/9781
-    payload.ext.wrapper.wiid = conf.wiid || bidderRequest.auctionId;
-    // eslint-disable-next-line no-undef
-    payload.ext.wrapper.wv = $$REPO_AND_VERSION$$;
-    payload.ext.wrapper.transactionId = conf.transactionId;
-    payload.ext.wrapper.wp = 'pbjs';
-    const allowAlternateBidder = bidderRequest ? bidderSettings.get(bidderRequest.bidderCode, 'allowAlternateBidderCodes') : undefined;
-    if (allowAlternateBidder !== undefined) {
-      payload.ext.marketplace = {};
-      if (bidderRequest && allowAlternateBidder == true) {
-        let allowedBiddersList = bidderSettings.get(bidderRequest.bidderCode, 'allowedAlternateBidderCodes');
-        if (isArray(allowedBiddersList)) {
-          allowedBiddersList = allowedBiddersList.map(val => val.trim().toLowerCase()).filter(val => !!val).filter(uniques);
-          biddersList = allowedBiddersList.includes('*') ? allBiddersList : [...biddersList, ...allowedBiddersList];
-        } else {
-          biddersList = allBiddersList;
-        }
-      }
-      payload.ext.marketplace.allowedbidders = biddersList.filter(uniques);
-    }
-
-    payload.user.gender = (conf.gender ? conf.gender.trim() : UNDEFINED);
-    payload.user.geo = {};
-    // TODO: fix lat and long to only come from request object, not params
-    payload.user.yob = _parseSlotParam('yob', conf.yob);
-    payload.site.page = conf.kadpageurl.trim() || payload.site.page.trim();
-    payload.site.domain = _getDomainFromURL(payload.site.page);
-
-    // add the content object from config in request
-    if (typeof config.getConfig('content') === 'object') {
-      payload.site.content = config.getConfig('content');
-    }
-
-    // merge the device from config.getConfig('device')
-    if (typeof config.getConfig('device') === 'object') {
-      payload.device = Object.assign(payload.device, config.getConfig('device'));
-    }
-
-    // update device.language to ISO-639-1-alpha-2 (2 character language)
-    payload.device.language = payload.device.language && payload.device.language.split('-')[0];
-
-    // passing transactionId in source.tid if present
-    if (bidderRequest?.ortb2?.source?.tid) {
-      deepSetValue(payload, 'source.tid', bidderRequest?.ortb2?.source?.tid);
-    }
-
-    // test bids
-    if (window.location.href.indexOf('pubmaticTest=true') !== -1) {
-      payload.test = 1;
-    }
-
-    // adding schain object
-    if (validBidRequests[0].schain) {
-      deepSetValue(payload, 'source.ext.schain', validBidRequests[0].schain);
-    }
-
-    // Attaching GDPR Consent Params
```

Comment:
In regard to the buildRequests function when using the ortbConverter library, i'm assuming that the ortbConverter lib handles properly setting kv's for privacy related params like gdpr, ccpa, gpp, coppa, etc.?  i'm guessing that this is all just abstracted away now?

Reply:
Yes thats correct @jlquaccia. This library is taking care of all these stuff.

Reply:
@kapil-tuptewar nice!  ok, that's what my assumption was, just wanted to confirm.  cool, all LGTM then.